src/BinSkim.Driver/AnalyzeOptions.cs

```diff
@@ -48,5 +48,10 @@ public class AnalyzeOptions : AnalyzeOptionsBase
             "ignorePdbLoadError",
             HelpText = "If enabled, BinSkim won't break if we have a 'PdbLoadingException'.")]
         public bool IgnorePdbLoadError { get; set; }
+
+        [Option(
+            "binskimAppInsightsKey",
+            HelpText = "An ApplicationInsightsKey to which will be send data. If not available, it will read from the environment variable.")]
+        public string BinskimAppInsightsKey { get; set; }
```

Comment:
>BinskimAppInsightsKey [](http://example.com/codeflow?start=22&length=21)

To facilitate the guardian integration, we will expose a new parameter and also be able to read the environment variable.

The reason: from azure pipelines, creating the variable it isn't passing the value correctly (the variable is always unavailable).

Reply:
We can't accept a change that encourages users to explicitly provide sensitive data on the command-line, this is a very poor security practice, as command-lines are persisted/logged in many contexts.

Reply:
thanks for the feedback.
i will remove the change and keep it clean to fix the issue that i faced.

Reply:
Just updated and sent an email with the details.
thanks for the help!